### DISCUSSION TITLE
Failed subscription payment

### DISCUSSION THREAD

**Discussion by sjelfull** [craftcms/commerce-stripe#167] [Ideas]
1) The plugin should handle failed payments (as far as I can see it doesn't process `invoice.payment_failed`) and trigger an event for this - and have some settings on what to do if a payment fails?

2) How do we send emails if a payment fails?

As an aside, what is the plans for supporting emails at all for subscriptions? Can't find anything in the docs.

**Comment by andris-sevcenko** [comment]
> The plugin should handle failed payments (as far as I can see it doesn't process invoice.payment_failed) and trigger an event for this - and have some settings on what to do if a payment fails?

This plugin gets notified when a subscription is expired by Stripe. How you handle the subscription retries can be managed on Stripe's dashboard.

Also, the plugin fires an `EVENT_RECEIVE_WEBHOOK`  event after doing everything it can for the webhooks, so you can tap into that.

I understand that it's not really a very helpful response and it doesn't seem very out-of-the-box, but before we start dealing with emails here, Commerce needs to re-work how it deals with emails and, IIRC, Commerce is waiting on Craft to change something there.

As it currently stands, we have to pick the battles and holes that can be reasonably plugged by plugins probably won't be first in line to get plugged.

I'm leaving this open as a feature request because it's a thing that should eventually be handled by this plugin.
